v93k/src/origen/origen/helpers/time.cpp

```diff
@@ -0,0 +1,84 @@
+/** @file */ 
+#include "../helpers.hpp"
+#include <cerrno>
+#include <stdlib.h>
+#include "../../origen.hpp"
+
+using namespace std;
+
+namespace Origen {
+
+namespace Time {
+
+static bool timeSet = false;
+static int _month;
+static int _day;
+static int _hour;
+static int _minute;
+static int _second;
+
+static time_t rawtime;
+static struct tm * timeinfo;
+
+
+void getTime(bool reset) {
+  time (&rawtime);
+  timeinfo = localtime (&rawtime);
+
+  if ((!timeSet) || reset) {
+    _month = (timeinfo->tm_mon + 1);
+    _day = timeinfo->tm_mday;
+    _hour = timeinfo->tm_hour;
+    _minute = timeinfo->tm_min;
+    _second = timeinfo->tm_sec;
+    timeSet = true;
+  }
+}
+
+/// Get the Month. If not set will query the time for any
+/// subsequent time type calls
+int month() {
+  if (!timeSet) {
+    getTime();
+    timeSet = true;
+  }
+  return _month;
+}
+
+/// Get the Day. Requires to run setTime prior to grab the time.
```

Comment:
These comments are not quite right, setTime seems to refer to a previous implementation?

Reply:
I had called it setTime before but then changed it yes.  I was debating last night whether I should change the name of the func back, since calling it does 'set' the time in the variables and never do you use it to get the time from the vars.. only to get the time from the system overall into origen-- so depends on how you look at it.  Which was makes the most sense to you?

Reply:
setTime would make more sense to me based on what it does, or maybe even lockTime or something like that

Reply:
initTime?

Reply:
Sounds good to me